tests/test_time.py

```diff
@@ -331,3 +331,50 @@ def test_max_lag_skip_years(self, max_lag, expected_index, expected_size):
             calendar.get_intervals().index.values, expected_index
         )
         assert calendar.get_intervals().iloc[0].size == expected_size
+
+
+class TestAnchorKwarg:
+    correct_inputs = [  # format: (anchor, anchor_fmt, anchor_str)
+        ("5-5", "%m-%d", "5-5"),
+        ("02-5", "%m-%d", "02-5"),
+        ("06-05", "%m-%d", "06-05"),
+        ("5-05", "%m-%d", "5-05"),
+        ("12-31", "%m-%d", "12-31"),
+        ("W01", "W%W-%w", "W01-1"),
+        ("W9", "W%W-%w", "W9-1"),
+        ("W01-4", "W%W-%w", "W01-4"),
+        ("W9-1", "W%W-%w", "W9-1"),
+        ("December", "%m", "12"),
+        ("dec", "%m", "12"),
+        ("jan", "%m", "1"),
+        ("Jan", "%m", "1"),
+    ]
+
+    incorrect_inputs = (
+        10,  # non-str inputs
+        (1, 2),
+        "0",  # month number less than 1
+        "13",  # month number greater than 12
+        "12-0",  # day number less than 1
+        "12-32",  # day number greater than 31
+        "31-12",  # incorrect month/day order
```

Comment:
One minor case, the combination of month abbreviation and digits (e.g. "Jan-12") is also not supported.

Reply:
You're right! I have added a test for month name + digit.